### DISCUSSION TITLE
System.ComponentModel.Composition vs. System.Composition

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/runtime#79492] [Q&A] [answered]
It seems some libraries have been overhauled or have successors, but it is not quite clear which one is "old" and which one should be preferred because there is no real official docs.

For example **MEF**, obviously there is **MEF1** in `System.ComponentModel.Composition` and there is **MEF2** in `System.Composition`. **Which one is the suggested framework?**

**The same is true for `Microsoft.Extensions.Configuration` and `System.Configuration`.**
It would be good if there would be some kind of categorization and below the category one can find the individual approaches. It would also be nice if the pros and cons are listed, since currently it's sometimes a problem. You start to evaluate the library and suddenly find, for example, that the configuration `Microsoft.Extensions.Configuration` cannot be saved, but only read.

Is there some official comparison documentation?

**Comment by teo-tsirpanis** [comment, marked as answer]
https://github.com/dotnet/runtime/blob/main/src/libraries/System.ComponentModel.Composition/README.md

**Reply by msedi** [reply]
@teo-tsirpanis. Thanks for pointing this out.
